cli/cli.go

```diff
@@ -439,123 +447,135 @@ func getMigrateSubcommand(subcmd, usage string) cli.Command {
-				log.Fatal("No schema specified in configuraion")
-			}
+	if len(forcedSchemas) > 0 {
+		modifiedSchemas = strings.Split(forcedSchemas, ",")
+	} else {
+		modifiedSchemas = migration.GetModifiedSchemas()
+	}
+	if len(modifiedSchemas) == 0 {
+		log.Info("No modified schemas, skipping post-migration event")
+		return
+	}
 
-			manager := schema.GetManager()
-			if err := manager.LoadSchemasFromFiles(schemaFiles...); err != nil {
-				log.Fatal(err)
-			}
+	log.Debug("Modified schemas: %s", modifiedSchemas)
 
-			rawDbConn, err := db.CreateFromConfig(config)
-			if err != nil {
-				log.Fatal(err)
-			}
-			dbConn := &server.DbSyncWrapper{DB: rawDbConn}
+	schemaFiles := config.GetStringList("schemas", nil)
+	if schemaFiles == nil {
+		log.Fatal("No schema specified in configuration")
+	}
 
-			sync, err := sync_util.CreateFromConfig(config)
-			if err != nil {
-				log.Fatal(err)
-			}
+	manager := schema.GetManager()
+	if err := manager.LoadSchemasFromFiles(schemaFiles...); err != nil {
+		log.Fatal(err)
+	}
 
-			identity, err := middleware.CreateIdentityServiceFromConfig(config)
-			if err != nil {
-				log.Fatal(err)
-			}
+	rawDbConn, err := db.CreateFromConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
+	dbConn := &server.DbSyncWrapper{DB: rawDbConn}
 
-			environmentManager := extension.GetManager()
+	sync, err := sync_util.CreateFromConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-			for _, s := range manager.Schemas() {
-				if !util.ContainsString(modifiedSchemas, s.ID) {
-					continue
-				}
-				pluralURL := s.GetPluralURL()
-
-				if _, ok := environmentManager.GetEnvironment(s.ID); !ok {
-					env := otto.NewEnvironment("post-migration-env", dbConn, identity, sync)
-					eventTimeout := context.Duration(PostMigrationEventTimeoutFlag)
-					env.SetEventTimeLimit(PostMigrationEvent, eventTimeout)
-					env.LoadExtensionsForPath(manager.Extensions, manager.TimeLimit, manager.TimeLimits, pluralURL)
-					log.Info("Loading environment for %s schema with URL: %s", s.ID, pluralURL)
-					if err != nil {
-						log.Fatal(fmt.Sprintf("[%s] %v", pluralURL, err))
-					}
-					environmentManager.RegisterEnvironment(s.ID, env)
-				}
+	identity, err := middleware.CreateIdentityServiceFromConfig(config)
+	if err != nil {
+		log.Fatal(err)
+	}
 
-				env, _ := environmentManager.GetEnvironment(s.ID)
-				eventContext := map[string]interface{}{}
-				eventContext["schema"] = s
-				eventContext["sync"] = sync
-				eventContext["db"] = dbConn
-				eventContext["identity_service"] = identity
-				err := env.HandleEvent(PostMigrationEvent, eventContext)
-				if err != nil {
-					log.Fatalf("Failed to handle event post-migration, err: %s", err)
-				}
-			}
-			etcdEvent := context.Bool(SyncEtcdEventFlag)
-			if !etcdEvent {
-				return
-			}
+	environmentManager := extension.GetManager()
 
-			syncWriter := server.NewSyncWriter(sync, dbConn)
-			_, err = syncWriter.Sync()
+	for _, s := range manager.Schemas() {
```

Comment:
this function is 93 LOC long. please split it. this for loop could be "handlePostMigrationInAllSchemas", the inner "if _, ok ..." could be getOrCreateEnvironment(schemaId string)

Reply:
ok, fixed

Reply:
ok, fixed

Reply:
is it fixed? publishEvent is still 70 LOC

Reply:
The function is now split in two parts, it is now 45 + 30 lines long.